icons/build-react.ts

```diff
@@ -46,147 +74,100 @@ const processFiles = (src: string, dest: string) => {
       // Will likely involve appending the lang/locale to the end of the friendly name for the unique component name
-      // const joinedDestPath = path.join(destPath, file)
+      // var joinedDestPath = path.join(destPath, file)
       // if (!fs.existsSync(joinedDestPath)) {
       //   fs.mkdirSync(joinedDestPath);
       // }
       // indexContents += processFolder(srcFile, joinedDestPath)
     } else {
-      if (oneSize && !file.includes("20")) {
-        continue;
+      if (resizable && !file.includes("20")) {
+        return;
       }
-      let iconName = file
-        // strip '.svg'
-        .substring(0, file.length - 4)
-        // strip ic_axis_
-        .replace("ic_axis_", "");
-      iconName = oneSize ? iconName.replace("20", "") : iconName;
-      // We want them to be camelCase, so access_time would become accessTime here
-      const destFilename = iconName
-        .split("_")
-        .map((p) =>
-          p.replace(p.substring(0, 1), p.substring(0, 1).toUpperCase())
-        )
-        .join("");
-
-      const iconContent = fs.readFileSync(srcFile, { encoding: "utf8" });
-
-      const jsxCode = oneSize
-        ? svgr.sync(iconContent, svgrOpts, { filePath: file })
-        : svgr.sync(iconContent, svgrOptsSizedIcons, {
-          filePath: file,
-        });
-      const jsCode = `
-const ${destFilename}Icon = (props: AxisIconProps) => {
-  const { fill: primaryFill = 'currentColor', className } = props;
-  return ${jsxCode};
-}
-
-export const ${destFilename} = /*#__PURE__*/wrapIcon(/*#__PURE__*/${destFilename}Icon, '${destFilename}');
-`;
+      var iconName = file.substr(0, file.length - 4); // strip '.svg'
+
+      iconName = iconName.replace("ic_axis_", ""); // strip ic_fluent_
+      iconName = resizable ? iconName.replace("20", "") : iconName;
+      var destFilename = _.camelCase(iconName); // We want them to be camelCase, so access_time would become accessTime here
+      destFilename = destFilename.replace(
+        destFilename.substring(0, 1),
+        destFilename.substring(0, 1).toUpperCase()
+      ); // capitalize the first letter
+
+      var iconContent = fs.readFileSync(srcFile, { encoding: "utf8" });
+      const getAttr = (key) =>
+        [...iconContent.matchAll(new RegExp(`(?<= ${key}=)".+?"`, "g"))].map(
+          (v) => v[0]
+        );
+      const width = resizable ? "\"1em\"" : getAttr("width")[0];
+      const paths = getAttr("d").join(",");
+      var jsCode =
+        `export const ${destFilename} = (/*#__PURE__*/createFluentIcon('${destFilename}', ${width}, [${paths}]));`;
       iconExports.push(jsCode);
     }
-  }
+  });
 
   // chunk all icons into separate files to keep build reasonably fast
+  /** @type string[][] */
   const iconChunks: string[][] = [];
   while (iconExports.length > 0) {
-    iconChunks.push(iconExports.splice(0, 500));
+    iconChunks.push(iconExports.splice(0, 1000));
```

Comment:
Interesting, does this mean they've increased the amount of icons in each chunk?

Reply:
Maybe they did it for a reason? 😅